### PR TITLE
Support GZip/Zlib compression and Shuffle filter for S3

### LINKED ISSUE #120: Add support for compressed and filtered datasets on S3 storage
Currently if a dataset is compressed or has any filters applied, PyActiveStorage will raise NotImplementedError in s3.py when using the S3 storage type.

There are many compression and filter algorithms supported by netCDF/HDF5, but this task covers supporting and testing the Zlib compression algorithm as well as the HDF5 Shuffle filter. This task covers support for S3 storage, building on the local storage support discussed in #118 and added in #119

### PR DESCRIPTION
**Pull request by markgoddard**
Adds support for netCDF files with Gzip/Zlib compression and/or the byte
shuffle filter when using the S3 storage backend.

The compression and filter integration tests now run when testing
against the S3 storage backend.

Closes #120

### CODE REVIEW

**Comment by markgoddard** [comment]
Please wait for #135 to merge before merging this PR.

File: activestorage/reductionist.py
Comment (valeriupredoi) [inline activestorage/reductionist.py:137]: we need to pull out the info in this docstring somewhere where it's a bit more visible too (README doubles as docs for now, and plop a link to the table)

**Review by valeriupredoi — APPROVED** [review]
ah great stuff, man, cheers muchly @markgoddard - one comment, and nothing else :beer:

**Pull request merged**
